src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.Accessor.cs

```diff
@@ -54,9 +53,16 @@ public Accessor(SQLitePersistentStorage storage)
             protected abstract void BindFirstParameter(SqlStatement statement, TDatabaseId dataId);
             protected abstract TWriteQueueKey GetWriteQueueKey(TKey key);
 
-            public async Task<Checksum> ReadChecksumAsync(TKey key, CancellationToken cancellationToken)
+            [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/36114", AllowCaptures = false)]
+            public Task<Checksum> ReadChecksumAsync(TKey key, CancellationToken cancellationToken)
+                => Storage.PerformReadAsync(s_readChecksum, (self: this, key, cancellationToken), cancellationToken);
+
+            private static readonly Func<(Accessor<TKey, TWriteQueueKey, TDatabaseId> self, TKey key, CancellationToken cancellationToken), Checksum> s_readChecksum =
+                t => t.self.ReadChecksum(t.key, t.cancellationToken);
```

Comment:
i'm using static fields now to ensure no accidental lambda captures.

Reply:
A lambda which just calls a static local function clearer

Reply:
but then we might accidentally capture in the lambda again, defeating the purpose of it.